instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/OpenTelemetryAppender.java

```diff
@@ -65,15 +77,32 @@ public void start() {
             captureMarkerAttribute,
             captureKeyValuePairAttributes,
             captureLoggerContext);
-    if (openTelemetry == null) {
-      openTelemetry = OpenTelemetry.noop();
-    }
     super.start();
   }
 
+  @SuppressWarnings("SystemOut")
   @Override
   protected void append(ILoggingEvent event) {
-    mapper.emit(openTelemetry.getLogsBridge(), event);
+    Lock readLock = lock.readLock();
+    readLock.lock();
+    try {
+      OpenTelemetry openTelemetry = this.openTelemetry;
+      if (openTelemetry != null) {
+        emit(openTelemetry, event);
```

Comment:
You could add an extra volatile read and `if` before locking (kinda like the double-checked lock pattern goes), and move the `emit()` call out of the locked section. This way after `openTelemetry` is set you won't need to obtain the lock, you'll just read a volatile (which you would have to do either way)

Reply:
pushed, I think it still needs the check inside the lock too(?), let me know what you think

Reply:
> pushed, I think it still needs the check inside the lock too(?), let me know what you think

Yup, it does need an extra check inside. LGTM, thanks!